### PR TITLE
update DateFormattingService

### PR DESCRIPTION
**Pull request by petropavel13**

### CODE REVIEW

File: Swift/ApiDateFormattingService.swift.twig [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+
+    typealias DateFormatType = ApiDateFormat
+
+    var currentRegion: Region = .Local()
 
-    func mappingTransform(for dateFormat: String) -> TransformOf<Date, String> {
-        return mappingTransform(for: DateFormattingArguments(dateFormat: dateFormat))
+    static let shared = ApiDateFormattingService()
+
+    private init() {
```
Comment (scoreyou) [inline Swift/ApiDateFormattingService.swift.twig]: what for?

**Review by scoreyou — CHANGES_REQUESTED** [review]

File: Swift/ApiDateFormattingService.swift.twig (hunk 1 above)
Comment (petropavel13) [reply Swift/ApiDateFormattingService.swift.twig]: private or init?)
Give me example where you need to subclass this service. Take in account that generated code will use ApiDateFormattingService, not any subclasses

Comment (scoreyou) [reply Swift/ApiDateFormattingService.swift.twig]: so `private init() {}` ?

Comment (petropavel13) [reply Swift/ApiDateFormattingService.swift.twig]: done

**Review by scoreyou — APPROVED** [review]

**Review by igorkislyuk — APPROVED** [review]

**Pull request merged**